### PR TITLE
[performance] cache media attachments

### PR DESCRIPTION
**Pull request by NyaaaWhatsUpDoc**
Changes:
- add media attachment caching
- slightly tweak default cache config values
- update all media attachment db calls to go through cache
- update envparsing test

Checklist:
- [x] I/we have read the [GoToSocial contribution guidelines](https://github.com/superseriousbusiness/gotosocial/blob/main/CONTRIBUTING.md).
- [x] I/we have discussed the proposed changes already, either in an issue on the repository, or in the Matrix chat.
- [x] I/we have performed a self-review of added code.
- [x] I/we have written code that is legible and maintainable by others.
- [x] I/we have commented the added code, particularly in hard-to-understand areas.
- [x] I/we have run tests and they pass locally with the changes.
- [x] I/we have run `go fmt ./...` and `golangci-lint run`.

### CODE REVIEW

**Comment by NyaaaWhatsUpDoc** [comment]
The main remaining issue here is that different testing areas are using different state.State instances (and therefore cache instances) which is causing the failures. *I think*.

**Comment by NyaaaWhatsUpDoc** [comment]
errors with "no space left on device" early on 🤔

File: internal/cache/gts.go [hunk 1]
```diff
@@ -145,6 +153,7 @@ func (c *gtsCaches) Stop() {
 	tryUntil("stopping gtsmodel.DomainBlock cache", 5, c.domainBlock.Stop)
 	tryUntil("stopping gtsmodel.Emoji cache", 5, c.emoji.Stop)
 	tryUntil("stopping gtsmodel.EmojiCategory cache", 5, c.emojiCategory.Stop)
+	tryUntil("stopping gtsmodel.MediaAttention cache", 5, c.media.Stop)
```
Comment (tsmethurst) [inline internal/cache/gts.go:156]: Media attention :D

File: internal/db/bundb/media.go [hunk 2]
```diff
@@ -51,22 +51,52 @@ func (m *mediaDB) GetAttachmentByID(ctx context.Context, id string) (*gtsmodel.M
...
+func (m *mediaDB) PutAttachment(ctx context.Context, media *gtsmodel.MediaAttachment) error {
+	return m.state.Caches.GTS.Media().Store(media, func() error {
+		_, err := m.conn.NewInsert().Model(media).Exec(ctx)
+		return m.conn.ProcessError(err)
+	})
+}
+
+func (m *mediaDB) UpdateAttachment(ctx context.Context, media *gtsmodel.MediaAttachment, columns ...string) error {
+	// Update the media's last-updated
+	media.UpdatedAt = time.Now()
```
Comment (tsmethurst) [inline internal/db/bundb/media.go:76]: It's possible here that if the caller provides columns which exclude last_updated, then the last updated value won't be set correctly. What I've been doing in other functions is checking if columns is set, and if so, adding 'last_updated' to the list by default. There's an example here: https://github.com/superseriousbusiness/gotosocial/blob/fe6c8b8152d723ff3788e35c79b2e840926d5b22/internal/db/bundb/status.go#L254-L258

There's likely a better way of doing this, however :P

File: internal/cache/gts.go (hunk 1 above)
Comment (NyaaaWhatsUpDoc) [reply internal/cache/gts.go:156]: ... :eyes:

File: internal/db/bundb/media.go (hunk 2 above)
Comment (NyaaaWhatsUpDoc) [reply internal/db/bundb/media.go:76]: ah gotcha shall handle this

**Pull request merged**